packages/mux-video/src/index.ts

```diff
@@ -377,27 +378,37 @@ class MuxVideoElement extends CustomVideoElement<HTMLVideoElement> implements Pa
     const video_title = this.getAttribute(Attributes.METADATA_VIDEO_TITLE);
     const viewer_user_id = this.getAttribute(Attributes.METADATA_VIEWER_USER_ID);
     return {
-      ...this.__metadata,
+      ...this.#metadata,
       ...(video_id != null ? { video_id } : {}),
       ...(video_title != null ? { video_title } : {}),
       ...(viewer_user_id != null ? { viewer_user_id } : {}),
     };
   }
 
   set metadata(val: Readonly<Metadata> | undefined) {
-    this.__metadata = val ?? {};
+    this.#metadata = val ?? {};
     if (!!this.mux) {
-      this.mux.emit('hb', this.__metadata);
+      this.mux.emit('hb', this.#metadata);
     }
   }
 
-  load() {
-    this.__core = initialize(this as Partial<MuxMediaProps>, this.nativeEl, this.__core);
+  async load() {
+    // Reset Hls.js and Mux data synchronously so all needed props are updated but
+    // wait 1 tick w/ loading the source to prevent multiple requests and cancels.
+    console.time('init core');
+    this.#core = initialize(this as Partial<MuxMediaProps>, this.nativeEl, this.#core);
```

Comment:
Is it an issue that we're potentially tearing down hls.js and mux-embed a bunch here?
If someone is fiddling with a bunch of props on the player, there might be a lot of churn. It's probably fine, since once it's settled and playback begins, we shouldn't be doing it anymore.

Reply:
I tested the performance hit and it's not that bad I think, it's under 1ms if there is no src.
I think it should be fine